include/alpaka/core/Concepts.hpp

```diff
@@ -25,20 +25,20 @@ namespace alpaka
         {
         };
 
-        namespace detail
-        {
+        template<
+            typename TConcept,
+            typename TDerived>
+        struct ImplementsConcept {
```

Comment:
Can you add some documentation to this trait now that it is public? Something like `Checks whether the concept is implemented by the given class`.